runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersBrowseView/UsersBrowseView.tsx

```diff
@@ -1,9 +1,83 @@
-import React, { FunctionComponent } from 'react';
+import React, { ChangeEvent, useMemo, VFC } from 'react';
 
-const UsersBrowseView: FunctionComponent = () => {
+import { Box, Tab, Tabs } from '@mui/material';
+import { useRouter } from 'next/router';
+import { FeatureKey } from 'runbotics-common';
+
+import InternalPage from '#src-app/components/pages/InternalPage';
+import If from '#src-app/components/utils/If';
+import useFeatureKey from '#src-app/hooks/useFeatureKey';
+import useTranslations from '#src-app/hooks/useTranslations';
+import { UsersTab } from '#src-app/utils/users-tab';
+
+import UsersListView from '../UsersListView';
+import UsersRegisterView from '../UsersRegisterView';
+
+import { StyledHeader } from './UsersBrowseView.styles';
+
+const UsersBrowseView: VFC = () => {
+    const router = useRouter();
+    const tab = router.pathname.match(/[^\/]+$/)[0];
+    const { translate } = useTranslations();
+    const hasUsersPageRead = useFeatureKey([FeatureKey.USERS_PAGE_READ]);
+
+    const UsersTabs = useMemo(() => {
+        const tabs = [];
+
+        if (hasUsersPageRead) {
+            tabs.push({
+                value: UsersTab.ALL_USERS,
+                label: translate('Users.Browse.Tabs.AllUsers.Label'
+                )});
+            tabs.push({
+                value: UsersTab.WAITING_USERS,
+                label: translate('Users.Browse.Tabs.Registration.Label'
+                )});
+        }
+
+        return tabs;
+    }, [hasUsersPageRead, translate]);
+
+    const handleTabChange = (event: ChangeEvent<HTMLInputElement>, value: UsersTab) => {
+        if (value === UsersTab.ALL_USERS) router.push('/app/users', null, { locale: router.locale });
+        else router.push('/app/users/pending', null, { locale: router.locale });
+    };
+
+    const getTabValue = () => 
+        tab === UsersTab.ALL_USERS ? UsersTab.ALL_USERS : UsersTab.WAITING_USERS;
+
+    const getPageTitle = () =>
+        getTabValue() === UsersTab.ALL_USERS
+            ? translate('Users.Browse.Tabs.AllUsers.Title')
+            : translate('Users.Browse.Tabs.Registration.Title'); 
 
     return (
-        <> Registration | Users </>
+        <> 
+            <InternalPage title={getPageTitle()}>
+                <StyledHeader/>
+                <Box>
+                    <If condition={hasUsersPageRead}>
```

Comment:
As preludium has noticed above if user don't have USERS_PAGE_READ and won't be able to even get into this page you can remove it. Check it with direct link. If so remove all uses